iroh-net/src/discovery/pkarr.rs

```diff
@@ -174,6 +183,54 @@ impl PublisherService {
     }
 }
 
+/// Resolve node info using a pkarr relay.
+///
+/// Pkarr stores signed DNS records in the mainline dht. These can be queried directly
+/// via the pkarr relay HTTP api or alternatively via a dns server that provides the
+/// pkarr records using `DnsDiscovery`. The main difference is that `DnsDiscovery` makes
+/// use of the system dns resolver and caching which can return stale records, while the
+/// `PkarrResolver` always gets recent data.
```

Comment:
so a relay does a lookup to the DHT on every request?

Reply:
+ LRU caching (available for all consumers of that relay and even PkarrClient that uses that domain as [resolver](pkarr.org/resolvers), and it also returns `cache-control` header, in case your HTTP client can use that.

Reply:
Maybe the documentation should say instead that the difference is that the relay returns a `SignedPacket` so it is trustless, unlike DNS.

Reply:
Good points. I don't think the ttl is used by iroh-net, seems like it should then. But ttl is a bit weird, if you have multiple DNS caches then the actual ttl is number of caches * ttl no?
The main reason for using this was what turned out to be a DNS misconfiguration (ttl for cache misses was set to 24h)

Reply:
PkarrClient uses the _smallest_ ttl in the packet, but it allows you to "clamp" that to a minimum or a maximum. So if your application prefers Zero TTL, it can do that, if it prefers at least 24h TTL it will cache packets for that long too.

But you can't control what clamp values the resolver/relay you are querying is using, only your client.

https://github.com/Nuhvi/pkarr/blob/main/pkarr/src/relay_client.rs#L29